### PR TITLE
[transfer-to-object] Turn on for mainnet

### PR DESCRIPTION
**Pull request by tzakian**
## Description 

Turns on transfer-to-object in mainnet.

## Test Plan 

Tested and has been running in devnet and testnet for the past couple months.

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [X] protocol change
- [X] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### Release notes

Enables transfer-to-object functionality on mainnet.

### CODE REVIEW

**Comment by lxfind** [comment]
btw I need to re-enable effects v2 first..

**Comment by tzakian** [comment]
> btw I need to re-enable effects v2 first..

This shouldn't be dependent on effects v2 I believe (but will double-check!). Shared object deletion does require it though so we'll need to wait for that one.

File: external-crates/move/crates/move-compiler/src/sui_mode/typing.rs [hunk 1]
```diff
@@ -888,6 +888,23 @@ fn exp(context: &mut Context, e: &T::Exp) {
             if is_transfer_module && PRIVATE_TRANSFER_FUNCTIONS.contains(&name.value()) {
                 check_private_transfer(context, e.exp.loc, mcall)
             }
+            // TODO(tzakian): Temporary
```
Comment (stefan-mysten) [inline external-crates/move/crates/move-compiler/src/sui_mode/typing.rs:891]: Is this part of the code well tested, if it's temporary? If not, is there a plan to address this in a follow-up PR?

**Review by amnn — APPROVED** [review]
I suspect this PR needs a rebase because it's based off an old version of #15531, but the tip looks good to me!

**Pull request merged**